libs/sdk-model/src/accessControl/index.ts

```diff
@@ -91,9 +119,108 @@ export const isUserGroupAccessGrantee = (obj: unknown): obj is IUserGroupAccessG
     return !isEmpty(obj) && (obj as IUserGroupAccessGrantee).type === "group";
 };
 
+/**
+ * Type of granted granular access.
+ *
+ * @public
+ */
+export type AccessGranularPermission = "VIEW" | "EDIT" | "SHARE";
+
+/**
+ * Access grantee specification with granular permissions.
+ *
+ * @public
+ */
+export interface IGranularAccessGrantee {
+    permissions: AccessGranularPermission[];
+    inheritedPermissions: AccessGranularPermission[];
+}
+
+/**
+ * User access grantee specification with granular permissions.
+ *
+ * @public
+ */
+export type IGranularUserAccessGrantee = IUserAccessGrantee & IGranularAccessGrantee;
+
+/**
+ * User group access grantee specification with granular permissions.
+ *
+ * @public
+ */
+export type IGranularUserGroupAccessGrantee = IUserGroupAccessGrantee & IGranularAccessGrantee;
+
+/**
+ * Tests whether the provided object is an instance of {@link IGranularAccessGrantee}.
+ *
+ * @param obj - object to test
+ * @alpha
+ */
+export const isGranularAccessGrantee = (obj: unknown): obj is IGranularAccessGrantee => {
+    return !isEmpty(obj) && (obj as IGranularAccessGrantee).permissions !== undefined;
```

Comment:
Again, this typeguard does not seems very safe :) Its code is equal to `isGranularAccess`, but it's testing for a different type

Reply:
Will update to distinguish the correct types here